src/trafficcontroller/internal/proxy/set_cookie_handler.go

```diff
@@ -15,6 +15,7 @@ func NewSetCookieHandler(domain string) *SetCookieHandler {
 func (h SetCookieHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	if err := r.ParseForm(); err != nil {
 		w.WriteHeader(http.StatusBadRequest)
+		return
```

Comment:
I am unsure if this will cause breaking changes to Apps Manager or other users of this endpoint. It almost seems as though this handler is serving dual purpose of setting a cookie and CORS headers. Usually your CORS headers would be handled in an HTTP `OPTIONS` request.